### PR TITLE
SimpleCov article

### PR DESCRIPTION
**Pull request by andycroll**

### CODE REVIEW

File: _posts/2017/2017-10-29-use-simplecov.markdown [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+title: 'Use simplecov in your Rails test suite'
+description: 'It’s not a silver bullet, but it is useful.'
+layout: article
+category: ruby
+image:
+  base: '2017/use-simplecov'
+  alt: 'Coverage HTML output'
+---
+
+A useful way of assessing the effectiveness of your testing is to use the [`simplecov` gem](https://github.com/colszowka/simplecov) with your Rails application to measure the lines of code that are exercised during your test suite.
```
Comment (nodunayo) [inline _posts/2017/2017-10-29-use-simplecov.markdown]: Long sentence. Can you split it into two?

Also `measure the lines of code that are exercised` isn't super clear. You're trying to say 'measure the lines of code that are run/touched', right?

File: _posts/2017/2017-10-29-use-simplecov.markdown [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+#### Add to `Gemfile`
+
+```ruby
+group :test do
+  gem 'simplecov', require: false
+end
+```
+
+And run `bundle` in the root directory of your project
```
Comment (nodunayo) [inline _posts/2017/2017-10-29-use-simplecov.markdown]: Add a full-stop to the end of 'project'.

File: _posts/2017/2017-10-29-use-simplecov.markdown [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+#### Add to `.gitignore`
+
+Do not commit generated files in the `coverage` directory.
+
+```
+coverage/*
+```
+
+#### At the top of `spec/spec_helper.rb` or `test/test_helper.rb`
```
Comment (nodunayo) [inline _posts/2017/2017-10-29-use-simplecov.markdown]: The other headings have all been commands: "Use", or "Add to..", etc. This one is different. Perhaps you want to make it the same format, e.g.: "Edit your `spec/spec_helper.rb`..."

File: _posts/2017/2017-10-29-use-simplecov.markdown [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+end
+```
+Add this right at the top of the file. The `add_filter` lines means files matching the passed string are excluded from the results.
+
+Then run your tests as normal and then open the `coverage/index.html` file that is created and you can explore your code.
+
+
+## But why?
+
+Measuring the coverage of your application’s code by your test suite is a blunt instrument. In fact, aiming for 100% coverage is not a noble aim, or even useful.
```
Comment (nodunayo) [inline _posts/2017/2017-10-29-use-simplecov.markdown]: Mmmm...I see what you're trying to say but it doesn't flow naturally from the "But why?"

Maybe start with some of the stuff you've got later down like: "Having an awareness..." and then later on talk about how aiming for 100% coverage is not a useful aim/not the point of the exercise.

File: _posts/2017/2017-10-29-use-simplecov.markdown [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+<blockquote class="blockquote">
+  <p class="mb-0">Some codebases with 100% test coverage will still have bugs, and some codebases with much lower coverage will not.</p>
+  <footer class="blockquote-footer">@sarahmei <cite title="Source Title"><a href="https://twitter.com/sarahmei/status/819270166576058369">11 Jan 2017</a></cite></footer>
+</blockquote>
+
+Having an awareness of your level of coverage is useful as an input to understanding the stability of your application and protection against regressions.
+
+And exploring the HTML reports is a good way to see if there are areas of your code that are poorly tested.
```
Comment (nodunayo) [inline _posts/2017/2017-10-29-use-simplecov.markdown]: I would remove the 'and' here.

File: _posts/2017/2017-10-29-use-simplecov.markdown [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+
+## Why not?
+
+It could slow your test suite a little. But it's only, at most, additional seconds in a multiple-minute test suite.
```
Comment (nodunayo) [inline _posts/2017/2017-10-29-use-simplecov.markdown:71]: Like this last sentence!

**Comment by andycroll** [comment]
@nodunayo Makes it better.

**Pull request merged**